lib/matplotlib/lines.py

```diff
@@ -790,8 +790,21 @@ def draw(self, renderer):
                     cap = self._dashcapstyle
                     join = self._dashjoinstyle
                 else:
+                    if tpath.should_simplify and affine.has_inverse:
+                        # the path simplification code expects the path
+                        # to be in pixel space, not data space, but the
+                        # renderer downstream expects the path to be
+                        # in data space. The transformation `affine`
+                        # will put the path into pixel space, and
+                        # affine.inverted() will transform it back to
+                        # data space. So, transform using affine, simplify
+                        # the path, and transform using the affine inverted.
+                        tpath = affine.inverted().transform_path(
+                            affine.transform_path(tpath).cleaned(simplify=True)
+                        )
```

Comment:
I'll play around with that. I'm a little worried about changing that since I don't know what downstream rendering code relies on (does it use the data-space to generate axes labels? scale text? etc.). It might be premature optimization at this point.